### ISSUE TITLE
Efficient regularization methods for classification

### ISSUE THREAD

**Issue by accosmin** [accosmin/nano#130]
Regularizing deep models takes too much time because several regularization factors have to be tried out. So remove the criterion_t part.

The idea would be to regularize in one shot:
* generate random samples with label {-1}+ (aka background images) and check if adding them to the training dataset improves performance
* randomly warp training images to extend the training set

Update benchmark_iterator:
* benchmark the impact of using different samplers (e.g. none, warp, nrand)

**Comment by accosmin** [comment]
Idea: null hyphotesis = by default predict no object {-1}. So pre-train the models on random tensors to predict {-1}.

**Comment by accosmin** [comment]
Random walk trainer: x(t+1) = x(t) - rand(0, 1) * g(t) with some smoothing and minibatch.

**Comment by accosmin** [comment]
Can regularize also as: sum(output(i) - {-1}^C) where C = #classes???
